main/acle.rst

```diff
@@ -404,6 +408,182 @@ Scalable Vector Extensions (SVE)
++==============================================================+
+| .. code:: c                                                  |
+|                                                              |
+|      int8x16_t svget_neonq[_s8](svint8_t vec)                |
++--------------------------------------------------------------+
+| .. code:: c                                                  |
+|                                                              |
+|      int16x8_t svget_neonq[_s16](svint16_t vec)              |
++--------------------------------------------------------------+
+| .. code:: c                                                  |
+|                                                              |
+|      int32x4_t svget_neonq[_s32](svint32_t vec)              |
++--------------------------------------------------------------+
+| .. code:: c                                                  |
+|                                                              |
+|      int64x2_t svget_neonq[_s64](svint64_t vec)              |
++--------------------------------------------------------------+
+| .. code:: c                                                  |
+|                                                              |
+|      uint8x16_t svget_neonq[_u8](svuint8_t vec)              |
++--------------------------------------------------------------+
+| .. code:: c                                                  |
+|                                                              |
+|      uint16x8_t svget_neonq[_u16](svuint16_t vec)            |
++--------------------------------------------------------------+
+| .. code:: c                                                  |
+|                                                              |
+|      uint32x4_t svget_neonq[_u32](svuint32_t vec)            |
++--------------------------------------------------------------+
+| .. code:: c                                                  |
+|                                                              |
+|      uint64x2_t svget_neonq[_u64](svuint64_t vec)            |
++--------------------------------------------------------------+
+| .. code:: c                                                  |
+|                                                              |
+|      float16x8_t svget_neonq[_f16](svfloat16_t vec)          |
++--------------------------------------------------------------+
+| .. code:: c                                                  |
+|                                                              |
+|      float32x4_t svget_neonq[_f32](svfloat32_t vec)          |
++--------------------------------------------------------------+
+| .. code:: c                                                  |
+|                                                              |
+|      float64x2_t svget_neonq[_f64](svfloat64_t vec)          |
++--------------------------------------------------------------+
+
+``svdup_neonq``
```

Comment:
I think we need a section in the design document explaining why we are adding these.

@rsandifo-arm wrote:

> It might be worth having “svdup_neonq” functions to duplicate a neon vector to fill an SVE one. This is just another way of writing svdupq_lane, without having to cast from Neon to SVE as a separate step. I see you've considered and rejected the possibility of extracting a Neon vector from an arbitrary point in the vector, and I think that's fair. The reasons I think the dup functions are nevertheless worthwhile are that:
>    * without it, the user would have to explicitly choose how to initialise the upper bits of the SVE register (in the svset_neonq call) only to overwrite those bits immediately.
>    * providing the function is consistent with svdup for scalar floating-point values, which are likewise implemented using the lane form of DUP.

Please summarize this in a new subsection in the design document.